### PR TITLE
Changes for S3 results in backend checks

### PR DESCRIPTION
**Pull request by MancunianSam**
The lambdas now need extra permissions to read and write to the results
bucket. There was a few bits that needed tidying, mostly concurrency
bits which were left over from the old checks.

### CODE REVIEW

File: root_backend_checks.tf [hunk 1]
```diff
@@ -221,21 +235,51 @@ module "statuses" {
...
+    "TDRYaraAVV2LambdaPolicy${title(local.environment)}" = templatefile("./templates/iam_policy/lambda_av_policy.json.tpl", {
+      function_name     = local.yara_av_v2_function_name,
+      account_id        = data.aws_caller_identity.current.account_id,
+      dirty_bucket      = local.upload_files_cloudfront_dirty_bucket_name
+      clean_bucket      = module.upload_bucket.s3_bucket_name
+      quarantine_bucket = module.upload_bucket_quarantine.s3_bucket_name
     })
   }
   role_name = "TDRYaraAVV2LambdaRole${title(local.environment)}"
+  runtime   = "python3.7"
```
Comment (TomJKing) [inline root_backend_checks.tf:253]: Is it worth putting the different lambda runtimes (java, python) as locals, for consistency. I see one of the lambdas is on python 3.9, this one is on 3.7?

File: root_backend_checks.tf [hunk 2]
```diff
@@ -250,9 +294,10 @@ module "backend_checks_step_function" {
   source             = "./tdr-terraform-modules/stepfunctions"
   tags               = local.common_tags
   project            = var.project
-  step_function_name = "TDRBackendChecks${title(local.environment)}"
+  step_function_name = "BackendChecks"
```
Comment (TomJKing) [inline root_backend_checks.tf:297]: Was there a reason for shortening this name?

File: templates/step_function/backend_checks_definition.json.tpl [hunk 3]
```diff
@@ -152,22 +145,66 @@
...
+      "Resource": "arn:aws:states:::lambda:invoke",
+      "OutputPath": "$.Payload",
+      "Parameters": {
+        "Payload.$": "$",
+        "FunctionName": "${backend_checks_results_arn}"
+      },
+      "Retry": [
+        {
+          "ErrorEquals": [
+            "Lambda.ServiceException",
```
Comment (TomJKing) [inline templates/step_function/backend_checks_definition.json.tpl:197]: Can these go in alphabetical order?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just some minor comments. Thanks

File: root_backend_checks.tf (hunk 1 above)
Comment (MancunianSam) [reply root_backend_checks.tf:253]: Yeah OK, I will at some point try to get everything on 3.9 but that's a future problem. I've put a ticket in Jira.

File: templates/step_function/backend_checks_definition.json.tpl (hunk 3 above)
Comment (MancunianSam) [reply templates/step_function/backend_checks_definition.json.tpl:197]: I almost got all of them in alphabetical order. I've updated these.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**